pkg/reconciler/instances/ory/action.go

```diff
@@ -107,23 +108,36 @@ func (a *postAction) Run(context *service.ActionContext) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to retrieve native Kubernetes GO client")
 	}
-	patchData, err := jwks.Get(jwksAlg, jwksBits)
+	secretObject, err := getSecret(context.Context, client, jwksNamespacedName)
 	if err != nil {
-		return errors.Wrap(err, "failed to generate JWKS secret")
+		return errors.Wrap(err, "Could not get JWKS secret")
 	}
-	if err := a.patchSecret(context.Context, client, jwksNamespacedName, patchData, logger); err != nil {
-		return errors.Wrap(err, "failed to patch Ory secret")
+
+	if isEmpty(secretObject) {
```

Comment:
why do we need to check if secret is empty?

Reply:
Because we create it in the Helm chart: https://github.com/kyma-project/kyma/blob/main/resources/ory/charts/oathkeeper/templates/secret.yaml

Reply:
My initial idea to create the secret beforehand would result in much more work needed to implement that.
Current Ory Oathkeeper helm chart requires this secret to be mounted as a volume.

Reply:
So we would need to also change the whole structure of Ory Reconciler.